### PR TITLE
Fix issue #56

### LINKED ISSUE #56: Missing soft-core parameters
We don't expose the `shift_coulomb` parameter and it appears that all soft-core parameters are not passed through to the dynamics object.

### PR DESCRIPTION
**Pull request by lohedges**
This PR closes #56  by adding the missing soft-core parameters to the config and passing all through to the dynamics object.

@akalpokas

### CODE REVIEW

**Comment by akalpokas** [comment]
Rerunning the GLU-->GLY perturbation with this fix is now causing the minimiser to hang for any simulations that are not λ=0.0 (any simulation where soft-core is being used).

**Comment by lohedges** [comment]
Well, I'm not surprised as the `somd1` settings are clearly not mitigating the nonbonded issues that I'm seeing elsewhere. Maybe try compiling the fix branch from [this](https://github.com/OpenBioSim/sire/pull/237) PR to see how you get on. There are still issues with the `ghost_14ff` that I'm trying to resolve.

**Comment by akalpokas** [comment]
No worries. The minimiser issue is fixed if you change the `coulomb_power` to 0 (sire's default - https://sire.openbiosim.org/tutorial/part07/03_ghosts.html#zacharias-softening) rather than `1.0` which is what SOMD2 is passing to sire dynamics in this PR. `coulomb_power` of 0 means that the `alpha` lever is no longer having an effect on the coulomb soft-core.

**Comment by lohedges** [comment]
We're using a `coulomb_power` of 1 to match the `somd1` settings, not the defaults from Sire. I think I've now fixed the soft-core issues. Will push another fix to my PR branch. It would be great if you could test when ready.

File: src/somd2/config/_config.py [hunk 1]
```diff
@@ -185,6 +186,10 @@ def __init__(
             Power to use for the soft-core Coulomb interaction. This is used
             to soften the electrostatic interaction.
 
+        shift_couloomb : str
```
Comment (mb2055) [inline src/somd2/config/_config.py:189]: `couloomb` should be `coulomb`

Comment (lohedges) [reply src/somd2/config/_config.py:189]: Nice to know that co-pilot make typos too :-)

**Review by mb2055 — APPROVED** [review]
Looks good :+1:

**Pull request merged**